walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,121 @@
+        
+        int event_start = 0;
+        int event_end = 0;
+
+        // Set of stirngs representing unique employees set to attend this meeting.
+        Set<String> event_attendees = null;
+
+        // Collection of string reprenting who is to attend the requedted meeting
+        Collection<String> request_attendees = request.getAttendees();
+        
+        // Boolean used to indictae whter or not the attendee should be ignored.
+        boolean shouldIgnore = true;
+
+        // Check if duration of event is longer than a whole day and if so return empty.
+        if ((int)request.getDuration() > TimeRange.WHOLE_DAY.duration()) {
+            return query;
+        }
+
+        // Check if there are any events happening at all if so return whole day query.
+        if (numberOfEvents == 0) {
+            // Adds a 24 hour time range to query.
+            query.add(TimeRange.WHOLE_DAY);
+            return query;
+        }
+        // Iterating throguh input event collection. 
+        for (Event event : events) {
+
+            // Collection of the pople who are to attend meeting.
+            attendees = event.getAttendees();
+
+            //Ignore event only if attendees of this event are not needed in requested meeting
+            for (String attendee : attendees) {
+                // If this attendee is required
+                if (request_attendees.contains(attendee)) {
+                    // You shouldn't ignore this attendee. 
+                    shouldIgnore = false;
+                    break;
+                } 
+                // If at the last event.
+                else if (eventCounter == numberOfEvents-1) {
+                    query.add(TimeRange.fromStartEnd(timeOfStart, timeOfEnd, true));
+                }
+            }
+            // if the attendee is mandatory.
+            if (!shouldIgnore) {
+                
+                // Fetch event start time.
+                event_start = event.getWhen().start();
+                // Fetch event end time.
+                event_end = event.getWhen().end();
+
+                if (event_start == timeOfStart) {
+                    timeOfStart = event_end;
+                }
+
+                // Check for overlapping/nested events
+                if (event_start < timeOfStart) {
+                    if (event_end > timeOfStart) {
+                        timeOfStart = event_end;
+                    }
+                }
+
+                //Non-overlapping/non-nested events
+                if (timeOfStart <= event_start) {
+                    if (event_start - timeOfStart >= desiredDuration) {
+                        query.add(TimeRange.fromStartEnd(timeOfStart, event_start, false));
+                    }
+                    timeOfStart = event_end;
+                }
+
+                // Last event of the day
+                if (eventCounter == numberOfEvents-1) {
+                    if (timeOfStart != timeOfEnd + 1) {
```

Comment:
This check is not clear to me, so maybe add a comment to explain here?

Reply:
These events counter first checks if the indeed is currently at the index representing the last event ( numberOfEvents - 1) if so then there is another check in which it ensures that the start time is not at midnight as timeOfEnd returns hours in minutes of 23:59 ( 1439) just to ensure that all the meetings are encased in one day.